### ISSUE TITLE
onClose event of DateTimePicker's Popover prevents closing with clicks on outside

### ISSUE THREAD

**Issue by mchingiz** [mantinedev/mantine#4076]
### What package has an issue

@mantine/dates

### Describe the bug

I want to listen to onClose event of DateTimePicker's Popover, so that I can show a confirmation dialog only after the user finalizes their date choice and closes the picker.

Trying to listen to `onClose` event of DateTimePicker's Popover, causes it to not close with clicks on outside. But it closes with 'submit' button. Same happens if DateTimePicker uses Modal instead of Popover

### What version of @mantine/hooks page do you have in package.json?

6.0.5

### If possible, please include a link to a codesandbox with the reproduced problem

https://codesandbox.io/s/epic-colden-fy5ums?file=/src/App.tsx

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

No

**Comment by rtivital** [comment]
Duplicate of #4105
